.github/workflows/build.yml

```diff
@@ -0,0 +1,72 @@
+name: pytorch-widedeep
+
+on:
+  push:
+#    branches: [ $default-branch ]
+  pull_request:
+#    branches: [ $default-branch ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: [3.7.9, 3.8, 3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install black flake8 pytest-cov codecov .
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Code Style (Black/Flake8)
+      run: |
+        # Black code style
+        black --check --diff pytorch_widedeep tests examples setup.py
+        # Stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E901,E999,F821,F822,F823 --ignore=E266 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --ignore=E203,E266,E501,E722,F401,F403,F405,F811,W503,C901 --statistics
+    - name: Test with pytest
+      run: |
+        pytest --doctest-modules pytorch_widedeep --cov-report html --cov-report term --disable-pytest-warnings --cov=pytorch_widedeep tests/
+    - name: Upload coverage
+      uses: actions/upload-artifact@v2
+      with:
+        name: coverage${{ matrix.python-version }}
+        path: .coverage
+
+  finish:
+    
+    needs: test
```

Comment:
I sent you the build file from a library called elegy, that is rather simple (I'd say) although they use poetry, but for this case I don't think it matters. 

Maybe the `pytest` line should be:

```python
pytest --doctest-modules pytorch_widedeep --cov-report xml --cov-report term --disable-pytest-warnings --cov=pytorch_widedeep tests/
```

only change is `xml` instead of `html`

In general the way they organise the tests is more readable? 

Let me know what you think, I am happy how it is now as long as we do not duplicate the Style tests. Simply, since we are using Actions, I thought we could write a proper clean `yml` file.

Reply:
I think the most important part of https://github.com/poets-ai/elegy/blob/master/.github/workflows/ci_test.yml is that they have separate codestyle job, as you mentioned in previous comment, ..., I just changed it to xml